### PR TITLE
Adding subtract to torch frontend

### LINKED ISSUE #13990: subtract

### PR DESCRIPTION
**Pull request by vtsamit**
Close #13990

### CODE REVIEW

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy/functional/frontends/torch/tensor.py [hunk 1]
```diff
@@ -205,6 +205,10 @@ def float(self, memory_format=None):
         cast_tensor.ivy_array = ivy.astype(self._ivy_array, ivy.float32)
         return cast_tensor
 
+    @with_unsupported_dtypes({"1.11.0 and below": ("bfloat16",)}, "torch")
+    def subtract(self, other, *, alpha=1):
+        return torch_frontend.subtract(self._ivy_array, other, alpha=alpha)
```
Comment (kurshakuz) [inline ivy/functional/frontends/torch/tensor.py:210]:
```suggestion
    @with_unsupported_dtypes({"1.11.0 and below": ("bfloat16",)}, "torch")
    def subtract(self, other, *, alpha=1):
        return torch_frontend.subtract(self, other, alpha=alpha)
```

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -2208,6 +2208,49 @@ def test_torch_instance_unsqueeze(
...
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="subtract",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=2,
+        min_value=-1e04,
+        max_value=1e04,
+        allow_inf=False,
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2221]: No need for this flag

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -2208,6 +2208,49 @@ def test_torch_instance_unsqueeze(
...
+    ),
+    alpha=st.floats(min_value=-1e04, max_value=1e04, allow_infinity=False),
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2223]: Similarly no need to set `allow_infinity=False` here

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 4]
```diff
@@ -2208,6 +2208,49 @@ def test_torch_instance_unsqueeze(
...
+# subtract
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="subtract",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=2,
+        min_value=-1e04,
+        max_value=1e04,
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2220]: You can remove setting those values as well

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 3 above)
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2223]: Similarly no need to add min and max values here

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 5]
```diff
@@ -2208,6 +2208,49 @@ def test_torch_instance_unsqueeze(
...
+        method_input_dtypes=input_dtype,
+        method_all_as_kwargs_np={
+            "other": x[1],
+            "alpha": alpha,
+        },
+        frontend_method_data=frontend_method_data,
+        init_flags=init_flags,
+        method_flags=method_flags,
+        frontend=frontend,
+        atol_=1e-02,
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2249]: Please remove that argument

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Thank you for your contribution! Implementation looks good to me, but please address the issues I mentioned

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
